### PR TITLE
fix/factor-out graphviz identifier encoding

### PR DESCRIPTION
**Pull request by mpenet**
graphviz will complain about "-" otherwise

I also had some lingering changes from when I was playing with it back when you announced it, I added them here too. I can remove them if you prefer. 

* makes linters happy about unused binding
* slightly more efficient transitions, invalid-rules checking

### CODE REVIEW

**Comment by mpenet** [comment]
Do you think it could be a good idea also to force qualified-keywords? for state/events/actions? In the spirit of "making everything unique/identifiable attributes"

File: src/automata/fsm.cljc [hunk 1]
```diff
@@ -21,17 +21,18 @@
...
          (dissoc ::actions)
          (assoc ::state to)
          (cond-> (some? actions) (assoc ::actions actions)))))
   ([rules state event]
    (let [transitions (get rules (extract state ::state))
          e           (extract event ::event)]
-     (or (first
-          (for [transition transitions :when (= (::event transition) e)]
-            transition))
+     (or (reduce #(when (identical? (::event %2) e)
```
Comment (pyr) [inline src/automata/fsm.cljc:34]: :+1:

File: src/automata/view.cljc [hunk 2]
```diff
@@ -2,11 +2,15 @@
   "Simplistic visualization builder. Leaves calling
    graphviz tools up to the caller")
 
+(defn identifier
+  [x]
+  (format "\"%s\"" (name x)))
```
Comment (pyr) [inline src/automata/view.cljc]: This will break cljc.
```suggestion
  (str \" (name x) \")))
```

**Review by pyr — APPROVED** [review]

**Comment by mpenet** [comment]
> Do you think it could be a good idea also to force qualified-keywords? for state/events/actions? In the spirit of "making everything unique/identifiable attributes"

Any thoughts on this? It's quite easy to mix event/state in the state table definition, that could help prevent this.

**Review by mcorbin — APPROVED** [review]

File: src/automata/fsm.cljc [hunk 3]
```diff
@@ -21,17 +21,18 @@
 
    Both arities throw when no possible transition was found"
   ([{::keys [rules state] :as machine} event]
-   (let [{::keys [actions to] :as transition} (transit rules state event)]
+   (let [{::keys [actions to] :as _transition} (transit rules state event)]
```
Comment (pyr) [inline src/automata/fsm.cljc]: seems as though `_transition` is never used, might as well remove it

**Review by pyr — COMMENTED** [review]
Another small comment on the code.

Regarding qualified keywords, do you mean to switch the `::state` and `::action` spec to enforce qualifed keywords?

File: src/automata/fsm.cljc (hunk 3 above)
Comment (mpenet) [reply src/automata/fsm.cljc]: Yes, that's why I put the underscore, I tought you put it for doc purposes. 
Apparently joker follows erlang conventions for this and ignores it when prefixed with _ that way.

**Comment by mpenet** [comment]
@pyr about qualified-keywords, yes exactly

**Comment by mpenet** [comment]
@pyr amended with changes discussed earlier.

**Comment by pyr** [comment]
Good to go

**Pull request merged**